app/src/main/java/com/arctouch/codechallenge/util/helpers/RetrofitHelper.kt

```diff
@@ -0,0 +1,44 @@
+package com.arctouch.codechallenge.util.helpers
+
+import android.util.Log
+import com.arctouch.codechallenge.BuildConfig
+import com.arctouch.codechallenge.api.TmdbApi
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
+
+
+object RetrofitHelper {
+
+    private var api: TmdbApi? = null
+
+    fun getApi(): TmdbApi? {
+        if (api == null) {
```

Comment:
I am using a lazy initialization, which means that it will be instantiated when its first needed. But, thinking about multi threading, a simple solution would be to initialize it statically, that way it would already exist once either thread needed it. Other ways would be to use other methods to lock the variables or methods and allow only one thread to access it at any given time. But considering a static class with static initialization I think it would be a simpler solution.